realm/src/main/java/io/realm/RealmFieldType.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import java.nio.ByteBuffer;
+
+import io.realm.dynamic.DynamicRealmObject;
+import io.realm.internal.Mixed;
+
+/**
+ * List of the types used by Realm's underlying storage engine.
+ *
+ * Normally there is no reason to interact with the underlying Realm types as Realm will automatically
+ * convert between normal Java types and the Realm types. However it is possible to access these
+ * types through a {@link DynamicRealmObject}.
+ */
+public enum RealmFieldType {
+    // Make sure numbers match with <realm/column_type.hpp>
+    INTEGER(0),
+    BOOLEAN(1),
+    STRING(2),
+    BINARY(4),
+    UNSUPPORTED_TABLE(5),
+    UNSUPPORTED_MIXED(6),
+    DATE(7),
+    FLOAT(9),
+    DOUBLE(10),
+    LINK(12),
```

Comment:
Then we have `RealmFieldType.REALMOBJECT`. Don't you guys think there are too many unnecessary `Realm`s?